source/core/serial_solid.cc

```diff
@@ -469,52 +467,55 @@ void
 SerialSolid<dim, spacedim>::write_output_results(
   std::shared_ptr<SimulationControl> simulation_control)
 {
-  DataOut<dim, spacedim> data_out;
-  data_out.attach_dof_handler(displacement_dh);
-
-  {
-    std::vector<std::string> displacement_names(spacedim, "displacement");
-    std::vector<DataComponentInterpretation::DataComponentInterpretation>
-      data_component_interpretation(
-        spacedim, DataComponentInterpretation::component_is_part_of_vector);
-
-    data_out.add_data_vector(displacement,
-                             displacement_names,
-                             DataOut<dim, spacedim>::type_dof_data,
-                             data_component_interpretation);
-  }
-
-  {
-    std::vector<std::string> displacement_names(spacedim,
-                                                "displacement_since_mapped");
-    std::vector<DataComponentInterpretation::DataComponentInterpretation>
-      data_component_interpretation(
-        spacedim, DataComponentInterpretation::component_is_part_of_vector);
-
-    data_out.add_data_vector(displacement_since_mapped,
-                             displacement_names,
-                             DataOut<dim, spacedim>::type_dof_data,
-                             data_component_interpretation);
-  }
-
-  data_out.build_patches();
-
-  const std::string folder        = simulation_control->get_output_path();
-  const std::string solution_name = simulation_control->get_output_name() +
-                                    ".solid_object." +
-                                    Utilities::int_to_string(id, 2);
-  const unsigned int iter        = simulation_control->get_step_number();
-  const double       time        = simulation_control->get_current_time();
-  const unsigned int group_files = simulation_control->get_group_files();
-
-  write_vtu_and_pvd<dim, spacedim>(pvdhandler,
-                                   data_out,
-                                   folder,
-                                   solution_name,
-                                   time,
-                                   iter,
-                                   group_files,
-                                   this->mpi_communicator);
+  if (*output_bool)
```

Comment:
instead I would have something like this
```suggestion
  if (!*output_bool) return;
```
You just leave the function if you are not outputting. Easier to read